integration/test/stubs.go

```diff
@@ -124,3 +133,52 @@ func AuthStub() *httptest.Server {
 		w.Write(resp)
 	}))
 }
+
+func KeycloakStub() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var resp []byte
+		if r.Method == http.MethodPost && r.RequestURI == "/realms/integration-test/protocol/openid-connect/token" {
+			if err := r.ParseForm(); err != nil {
+				w.WriteHeader(http.StatusNotImplemented)
+			} else {
```

Comment:
if you return after the status not implemented, you can dedent the rest of this, and make the logic a little easier to read.

Reply:
Is the call to `w.Write()` not needed? I believe it is.

Reply:
It's only needed if there's a response body, and in this case, I don't believe there is.